Use Pydantic settings

joejoinerr opened the pull request:
I went ahead and added this anyway for you to explore, but of course you don't have to use. 

Pydantic settings has all of the functionality that you were writing manually. Plus you also get more resilience since all of the values are type-coerced and validated. Defaults can also be provided, as I have with some of them. Therefore you don't have to check if a key exists or it is what you expect it to be in the program - you can just go ahead and use it.

joejoinerr committed:
Replace DotDict settings with Pydantic

jroakes commented:
@joejoinerr This looks good.  Will you check off this in Readme :-)?   Use something called Pydantic (Hey Joe!)
Also, please double confirm this is tested and config in all files were updated.

joejoinerr committed:
Update readme

joejoinerr commented:
I updated the readme. I have updated all the config references I can see

I also ran a sitemap_test and email_test with the Loco data already in there and both finished without errors 👍 Can you check the email?
Are there any more that you want me to run?

jroakes commented:
Thank you!
